packages/invoices/src/OutboundShipment/ListView/api.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  ObjectWithStringKeys,
+  InvoicesQuery,
+  request,
+  gql,
+  batchRequests,
+  SortBy,
+  ListApi,
+  Invoice,
+  GraphQLClient,
+  getSdk,
+  InvoiceSortFieldInput,
+  InvoiceRow,
+  InvoicePriceResponse,
+} from '@openmsupply-client/common';
+import { Environment } from '@openmsupply-client/config';
+
+const client = new GraphQLClient(Environment.API_URL);
+const api = getSdk(client);
+
+const pricingGuard = (pricing: InvoicePriceResponse) => {
+  if (pricing.__typename === 'InvoicePricingNode') {
+    return pricing;
+  } else if (pricing.__typename === 'NodeError') {
+    throw new Error(pricing.error.description);
+  } else {
+    throw new Error('Unknown');
+  }
+};
```

Comment:
so I put these guards (and some types have optional `__typename` and I can't figure out why, but thats the unknown error..)

Maybe it's wrong but I figure if there's been an error fetching the invoices 'totalAfterTax' then there's some error in the server .. 🤷  - but maybe I should instead be returning some different value

Reply:
I'd prefer not to have too much logic in the front end, trying to interpret errors and behave differently depending on exactly what has gone wrong. Useful in some specific cases, but as a general rule - I think better to let the user know that things are just plain broke.

So yes - I agree. Just throw the error